### PR TITLE
WIP: Improve CustomResource class for Quarkus

### PR DESCRIPTION
**Pull request by metacosm**
This PR is built on top of #2776 so most commits will go away once that PR gets merged and this one is rebased.

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/CustomResource.java [hunk 1]
```diff
@@ -71,11 +65,10 @@
   "status"
 })
 @Buildable(builderPackage = "io.fabric8.kubernetes.api.builder", editableEnabled = false)
-public abstract class CustomResource<S, T> implements HasMetadata {
+public abstract class CustomResource<S, T> implements HasMetadata, CustomResourceMetadata<S, T> {
```
Comment (jorsol) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/CustomResource.java:68]: If the CustomResourceMetadata extends HasMetadata, it should not be necessary to duplicate HasMetadata here.

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/CustomResourceMetadata.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+  }
+
+  @JsonIgnore
+  default boolean isStorage() {
+    return getStorage(getClass());
+  }
+
+  S getSpec();
+
+  T getStatus();
```
Comment (jorsol) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/CustomResourceMetadata.java:165]: What is the use of having getSpec() and getStatus() here? it's not enough to have them in the abstract class?

I think having them here defeats the purpose of a "Metadata" type.

**Comment by manusa** [comment]
Is this PR still relevant?

**Comment by metacosm** [comment]
This is more scoped for later and might be replaced by some other improvements but I'd rather keep it open for now.

**Comment by manusa** [comment]
:+1: let's keep it then

**Comment by metacosm** [comment]
Closing this for now, will revisit if needed later.

**Pull request closed without merging**